contracts/StrategyCurveTwoPool.sol

```diff
@@ -0,0 +1,383 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+// These are the core Yearn libraries
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+import { ICurveGauge, ICurveGaugeFactory, ICurvePool } from "../interfaces/ICurveTwoPool.sol";
+import { ISwapRouter } from "../interfaces/ISwapRouter.sol"; // Uni V3
+import { BaseStrategy, StrategyParams } from "@yearnvaults/contracts/BaseStrategy.sol";
+
+// To avoid the "stack too deep" error
+struct ContractAddress {
+  address vault;
+  address healthCheck;
+  address usdt;
+  address usdc;
+  address weth;
+  address crv;
+  address router;
+  address gauge;
+  address gaugeFactory;
+  address pool;
+}
+
+abstract contract StrategyCurveBase is BaseStrategy {
+  using Address for address;
+
+  /* ========== STATE VARIABLES ========== */
+  // these should stay the same across different wants (`2CRV` LP tokens, `crv3crypto` LP tokens, etc.).
+
+  ICurveGauge public immutable gauge;
+
+  ICurveGaugeFactory public immutable gaugeFactory;
+
+  ICurvePool public immutable curve;
+
+  address public immutable wethAddress;
+
+  // Uniswap V3 router
+  ISwapRouter public immutable router;
+
+  IERC20 public crv;
+
+  bool internal forceHarvestTriggerOnce; // only set this to true externally when we want to trigger our keepers to harvest for us
+  uint256 public creditThreshold; // amount of credit in underlying tokens that will automatically trigger a harvest
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor(ContractAddress memory _contractsAddress)
+    public
+    BaseStrategy(_contractsAddress.vault)
+  {
+    // Curve contracts
+    gauge = ICurveGauge(_contractsAddress.gauge);
+    gaugeFactory = ICurveGaugeFactory(_contractsAddress.gaugeFactory);
+    curve = ICurvePool(_contractsAddress.pool);
+    crv = IERC20(_contractsAddress.crv);
+
+    // Uniswap V3 router
+    router = ISwapRouter(_contractsAddress.router);
+
+    // Wrapped ETH - Used for multi-hop swap. No contract instance needed
+    wethAddress = _contractsAddress.weth;
+
+    // Arbitrum healthcheck contract address
+    healthCheck = _contractsAddress.healthCheck;
+
+    // Our standard approvals. want = Curve LP token
```

Comment:
Linting needed here and elsewhere

Reply:
yeah, sorry. I'll fix it. I noticed the original source code had tab defined with 4 spaces and in my starting setup it was defined with 2. I seems that the linter didn't like me updating to 4 spaces somewhere between commits.